config/environments/test.rb

```diff
@@ -4,7 +4,7 @@
   # Some tests require the following languages (en, es, fr, nl, pt-BR)
   # so we override the available languages for the test environment.
   config.i18n.default_locale = :en
-  config.i18n.available_locales = %w[de en es fr nl pt-BR zh-CN]
+  config.i18n.available_locales = %w[ar de en es fr nl pt-BR zh-CN]
```

Comment:
Including extra locales in the tests is tricky because the pages loading translatable fields become slower and so does the test suite :thinking:. Some of the tests even fail sometimes because loading the page takes too long.

How about removing this change and, in the test, doing:

```ruby
    let!(:default_enforce) { I18n.enforce_available_locales }

    before do
      I18n.enforce_available_locales = false
      allow(I18n).to receive(:available_locales).and_return(%i[ar en es fr])
    end

    after { I18n.enforce_available_locales = default_enforce }
```

However, this would require changing the tests so we write `expect(page).to have_link href: "/?locale=ar"` instead of `expect(page).to have_link "عربى", href: "/?locale=ar"`. If we want to keep the test as it is, we'd have to write something like:

```ruby
    let!(:default_locales) { I18n.available_locales }

    before do
      I18n.available_locales = %i[ar en es fr]
      I18n.reload!
    end

    after do
      I18n.available_locales = default_locales
      I18n.reload!
    end
```

Since reloading the I18n backend takes a second or two, if we follow this approach, I'd suggest doing this just for the test we've added in this pull request, and not for the "renders a list of links" test.

What do you think? :thinking:

Come to think about it, there might be a couple of languages we could remove from this array and then enable them only when needed and that might speed up some tests :thinking:. I'll take that into consideration :wiggum:.

Reply:
Hi @javierm, I agree with you! I finally opted for the second option and created a new context in the component test, so it does not affect other tests. 

>Come to think about it, there might be a couple of languages we could remove from this array and then enable them only when needed and that might speed up some tests 🤔. I'll take that into consideration :wiggum:.

Are you talking about the `available_locales` defined at `config/environments/test.rb` file? 🤔 Probably we can remove some from there and speed up the suite a bit. I annotated that into my Wigum typing machine 😉 .